File: hotel-app/src/main/kotlin/eu/getsoftware/hotelico/hotelapp/application/hotel/common/dto/HotelDTO.java
package eu.getsoftware.hotelico.hotelapp.application.hotel.common.dto;

import eu.getsoftware.hotelico.clients.common.dto.BasicDTO;
import eu.getsoftware.hotelico.clients.common.utils.ControllerUtils;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.Setter;
import lombok.With;
import lombok.experimental.SuperBuilder;

@With
@Getter
@SuperBuilder()
public class HotelDTO extends BasicDTO
{
    private int rating = 0;
    
    private int activityNumber = 0;
    
    private int customerNumber = 0;
    
    private int anonymeGuestNumber = 0;
    
    private Long consistencyId = 0L;

    @Setter(AccessLevel.NONE)
    private long creationTime = 0L;

    private double kmFromMe = 0.0;
    
    private String name = null;
    
    private String wellcomeMessage = null;
    
    private String description = null;
    
    private String info = null;
    
    private String city = null;
    
    private String postalCode = null;
    
    private String street = null;
    
    private String house = null;
    
    private String phone = null;
    
    private String fax = null;
    
    private String email = null;
    
    private String currentHotelAccessCode= null;

//    private  String picture;

    private  String pictureUrl= null;

    private  String previewPictureUrl= null;
    
    private String website = null;
    
    private String mapLink = null;
    
    private Boolean virtualfalse;
    
    public HotelDTO(long initId){
        super(initId);
    }
    
    public long getCreationTime() {
      return (creationTime > 0)? creationTime : getId();
   }
    
    public String getPictureUrl() {
      return ControllerUtils.addHostPrefixOnDemand(pictureUrl);
   }
    
    public String getPreviewPictureUrl() {
        
        String picUrl = (ControllerUtils.isEmptyString(previewPictureUrl))? ControllerUtils.PREVIEW_HOTEL_NOT_AVAILABLE_URL : previewPictureUrl;

        return ControllerUtils.addHostPrefixOnDemand(picUrl);
   }
    
    public void setCustomerNumber(int customerNumber)
    {
        this.customerNumber = customerNumber;
    }
    
    public void setCreationTime(long creationTime)
    {
        this.creationTime = creationTime;
    }
    
    public void setAnonymeGuestNumber(int anonymeGuestNumber)
    {
        this.anonymeGuestNumber = anonymeGuestNumber;
    }
    
    public void setActivityNumber(int activityNumber)
    {
        this.activityNumber = activityNumber;
    }
    
    public void setKmFromMe(double kmFromMe)
    {
        this.kmFromMe = kmFromMe;
    }
}

File: src/main/kotlin/de/hotelico/dto/HotelDTO.kt
package de.hotelico.dto;

import de.hotelico.utils.ControllerUtils;

public class HotelDTO: BasicDTO
{
    var rating: Int = 0

    var activityNumber = 0

    var customerNumber = 0

    var anonymeGuestNumber = 0

    var consistencyId: Long? = 0L

    private var creationTime: Long = 0L

    var kmFromMe = 0.0

    var name: String? = null

    var wellcomeMessage: String? = null

    var description: String? = null

    var info: String? = null

    var city: String? = null

    var postalCode: String? = null

    var street: String? = null

    var house: String? = null

    var phone: String? = null

    var fax: String? = null

    var email: String? = null

    var currentHotelAccessCode: String? = null

//    private String picture;

    private var pictureUrl: String? = null

    private var previewPictureUrl: String? = null

    var website: String? = null

    var mapLink: String? = null

    var virtual: Boolean = false

   constructor(): super()
   

   fun getCreationTime(): Long? {
      return if (creationTime > 0) creationTime else id.toLong()
   }

   fun setCreationTime(creationTime: Long) {
      this.creationTime = creationTime
   }
   
   fun getPictureUrl(): String? {
      return ControllerUtils.addHostPrefixOnDemand(pictureUrl)
   }

   fun setPictureUrl(pictureUrl: String) {
      this.pictureUrl = pictureUrl
   }

   fun getPreviewPictureUrl(): String? {
      val picUrl = if (ControllerUtils.isEmptyString(previewPictureUrl)) ControllerUtils.PREVIEW_HOTEL_NOT_AVAILABLE_URL else previewPictureUrl

      return ControllerUtils.addHostPrefixOnDemand(picUrl)
   }

   fun setPreviewPictureUrl(previewPictureUrl: String) {
      this.previewPictureUrl = previewPictureUrl
   }
}
